operator/pkg/controllers/infrastructure/v1alpha1/controller/controlplane.go

```diff
@@ -0,0 +1,76 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/awslabs/kit/operator/pkg/apis/infrastructure/v1alpha1"
+	"github.com/awslabs/kit/operator/pkg/awsprovider"
+	"github.com/awslabs/kit/operator/pkg/controllers"
+	"github.com/awslabs/kit/operator/pkg/status"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type controlPlane struct {
+	ec2api *awsprovider.EC2
+	client.Client
+	etcdprovider *etcdProvider
+}
+
+// NewControlPlaneController returns a controller for managing VPCs in AWS
+func NewControlPlaneController(ec2api *awsprovider.EC2, kubeClient client.Client) *controlPlane {
```

Comment:
imo, you should nest these packages such that you can call

controlplane.NewController.

On another note, I'd get rid of this v1alpha1 notion in your folder structures.

I think we could simplify with something like
```
pkg/controllers/controlplane/controller.go
```